README.md

```diff
@@ -116,60 +116,83 @@ In the absence of a `channels` key, some sensible defaults for RAPIDS will be us
+
+#### `common` Key
+
+The `common` key contains a list of objects with the following keys:
+
+- `output_types` - a list of output types (e.g. "conda" for `environment.yaml` files or "requirements" for `requirements.txt` files) for which the packages in the `packages` key should be applied to
+- `packages` - a list of packages to be included in the generated output file
+
+#### `specific` Key
+
+The `specific` key contains a list of objects with the following keys:
+
+- `output_types` - _same as `output_types` for the `common` key above_
+- `matrices` - a list of objects (described below) which define packages that are specific to a particular matrix combination
 
-The `common` key has children which are simple key-value pairs, where the key is an arbitrary name for the dependency list (i.e. `build`, `test`, `libcuml_build`, `cuml_build`, etc.) and the value is a list of dependencies.
+##### `matrices` Key
 
-The `specific` key's value is an array of objects. Each object contains a `matrix` key and some arbitrarily named dependency lists (similar to the dependency lists under `common`). The `matrix` key is used to define which matrix combinations from `files.[*].matrix` these dependency lists should apply to. This is elaborated on in [How Dependency Lists Are Merged](#how-dependency-lists-are-merged)
+Each list item under the `matrices` key contains a `matrix` key and a `packages` key. The `matrix` key is used to define which matrix combinations from `files.[*].matrix` these dependency lists should apply to. This is elaborated on in [How Dependency Lists Are Merged](#how-dependency-lists-are-merged). The `packages` key is a list of packages to be included in the generated output file
 
 An example of the above structure is exemplified below:
 
 ```yaml
 dependencies:
-  - output_types: [conda, requirements] # common dependencies between conda environment.yaml & requirements.txt files
-    common: # common between all matrix variations
-      build: # arbitrarily named dependency list
-        - common_build_dep
-      test: # arbitrarily named dependency list
-        - pytest
-    specific:
-      # dependencies specific to x86_64 and 11.5
-      - matrix:
-          cuda: "11.5"
-          arch: x86_64
-        build:
-          - a_random_x86_115_specific_dep
-  - output_types: [conda] # dependencies specific to conda environment.yaml files
+  build: # dependency list name
+    common: # dependencies common among all matrix variations
+      - output_types: [conda, requirements] # the output types this list item should apply to
+        packages:
+          - common_build_dep
+      - output_types: conda
+        packages:
+          - cupy
+          - pip: # supports `pip` key for conda environment.yaml files
+              - some_random_dep
+    specific: # dependencies specific to a particular matrix combination
+      - output_types: [conda, requirements]
+        matrices:
+          - matrix: # dependencies specific to x86_64 and 11.5
```

Comment:
Do we need an empty matrix below this, to indicate what happens for `not x86_64 or not cuda==11.5`?